core/core.js

```diff
@@ -9,12 +9,53 @@
   let opsCache = {};
   const errorMap = {};
   let nextPromiseId = 1;
-  const promiseTable = new Map();
+  const promiseMap = new Map();
+  const RING_SIZE = 4 * 1024;
+  const NO_PROMISE = null; // Alias to null is faster than plain nulls
```

Comment:
What, really??!

Reply:
Yeah, it appears to save ~10ns/opcall in the benches I ran.

I haven't double-checked the v8 generated bytecode, but I assume it could be faster due to checking equality via referential equality instead of value equality.

Reply:
I will definitely need to double check this and narrow it down, to see if there is a true performance difference or if was just a statistical fluke in my local benches.

But it's easy to change either way and doesn't negatively impact performance or correctness in its current form.

Reply:
@piscisaureus I've dumped the v8 bytecode and diffed the bytecode implementations of `getPromise()` and `setPromise()`

## setPromise()

![image](https://user-images.githubusercontent.com/949223/113860978-f83c5180-97a6-11eb-96ee-03eb9ebaa9d5.png)

## getPromise()

![image](https://user-images.githubusercontent.com/949223/113861026-07bb9a80-97a7-11eb-9de6-ee39788c24ab.png)